### PR TITLE
Raiders search through buildings

### LINKED ISSUE #9454: [BUG] <title>
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Are you using the latest MineColonies Verison?

- [X] I am running the latest **alpha** version of MineColonies for my Minecraft version.

### Did you check on the Wiki? or ask on Discord?

- [X] I checked the MineColonies Wiki and made sure my issue is not covered there. Or I was sent from discord to open an issue here.

### Minecraft Version

1.2

### MineColonies Version

idk its ATM9 0.1.12

### Structurize Version

idk its ATM9 0.1.12

### Related Mods and their Versions

image

### Reproduction Steps

_No response_

### Logs

https://gist.github.com/romamuravey/085b7c0b2a7cd6251a5d555fd01ded0a/56af004b158fc9034470538eefb9812d3c8a2213

### Anything else?

* Add a thumbs-up to the bug report if you are also affected. This helps the bug report become more visible to the team and doesn't clutter the comments.
* Add a comment if you have any insights or background information that isn't already part of the conversation.

### LINKED ISSUE #9472: [BUG] Mercy Ends to Raids still say "Victory"
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Are you using the latest MineColonies Version?

- [X] I am running the latest **alpha** version of MineColonies for my Minecraft version.

### Did you check on the Wiki? or ask on Discord?

- [X] I checked the MineColonies Wiki and made sure my issue is not covered there. Or I was sent from discord to open an issue here.

### What were you playing at the time? Were you able to reproduce it in both settings?

- [ ] Single Player
- [X] Multi Player

### Minecraft Version

1.20.1

### MineColonies Version

Ray: "Almost the newest"

### Structurize Version

Ray: "Almost the newest"

### Related Mods and their Versions

_No response_

### Current Behavior

When a raid kills enough citizens or guards, the raid automatically ends. The ending message says "Victory, the colony was successfully defended!"

Note: I only saw this in the stream. I didn't personally experience this.

### Expected Behavior

It should say some message that the colony was defeated and the raiders pillaged enough or something. Ask Thodor.

### Reproduction Steps

Do the raid.
Let everyone die.
Read chat.

### Logs

N/A

### Anything else?

* Add a thumbs-up to the bug report if you are also affected. This helps the bug report become more visible to the team and doesn't clutter the comments.
* Add a comment if you have any insights or background information that isn't already part of the conversation.

### LINKED ISSUE #9551: [BUG] 1.20.1/+ When Destroying Spawners on Pirate Ships the Raid Should End According to Set Number of Days In Config
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Are you using the latest MineColonies Version?

- [X] I am running the latest **alpha** version of MineColonies for my Minecraft version.

### Did you check on the Wiki? or ask on Discord?

- [X] I checked the MineColonies Wiki and made sure my issue is not covered there. Or I was sent from discord to open an issue here.

### What were you playing at the time? Were you able to reproduce it in both settings?

- [X] Single Player
- [X] Multi Player

### Minecraft Version

1.20.1

### MineColonies Version

minecolonies-1.20.1-1.1.303-BETA.jar

### Structurize Version

 structurize-1.20.1-1.0.659-BETA.jar

### Related Mods and their Versions

_No response_

### Current Behavior

Even though I have set in config for the pirate ship to be set to disappear in 24 hours they remain. I was told its because I am destroying spawners. Ironically why should this matter? At any rate if it does matter it should be fixed became it makes your town do nothing for 3 days.

### Expected Behavior

Regardless to how a player removes threats your town should go back to its business. if Actual pilgrims took this stance they would have died. I feel like days present should only apply to living threats and once they are dealt with this should return to normal with no delay at all. I play with dangerous mobs in pack and I enjoy a challenge but I do not enjoy when colonists commit suicide and do nothing. Or a threat arrives and they do nothing. This honestly seems very; very stupid. 

### Reproduction Steps

Set Config to 1 day
Get raided by pirates and then go burn down or destroy ships.

### Logs

NA

### Anything else?

Again as stated I play with leveling mobs and monsters and the raids add extra spice and is meant to be forced to deal with; I even have it set to be max difficulty and spawn rate - what I don't want to do is be punished for successfully defending my colony or colonies. Thanks for your time.

### PR DESCRIPTION
**Pull request by Raycoms**
Closes #9551
Closes #9472
Closes #9454

# Changes proposed in this pull request:
- Use the right spawn event
- Make env damage 0 until they walked away
- Make raiders walk around in buildings after arrival for a bit.
- Make citizens hide when raiders pass through their colony.
- Make citizens stop hiding when the ship is defeated but didnt sail away yet
- Print a different message on mercy end.
- Dont crash on custom arrow injection

[ x ] Yes I tested this before submitting it.
[ ] I also did a multiplayer test.

Review please

### CODE REVIEW

File: src/main/java/com/minecolonies/coremod/entity/mobs/aitasks/RaiderWalkAI.java [hunk 1]
```diff
@@ -64,8 +80,9 @@ private boolean walk()
                 walkToCampFire();
                 return false;
             }
+            raider.setTempEnvDamageImmunity(false);
 
-            if (targetBlock == null || raider.blockPosition().distSqr(targetBlock) < 25 || raider.level.getGameTime() > walkTimer)
+            if (targetBlock == null || raider.level.getGameTime() > walkTimer)
             {
                 targetBlock = raider.getColony().getRaiderManager().getRandomBuilding();
```
Comment (someaddons) [inline src/main/java/com/minecolonies/coremod/entity/mobs/aitasks/RaiderWalkAI.java:88]: reset walkInBuildingState to false in here

File: src/main/java/com/minecolonies/coremod/entity/mobs/aitasks/RaiderWalkAI.java [hunk 2]
```diff
@@ -74,6 +91,32 @@ private boolean walk()
...
+                        {
+                            randomPathResult = null;
+                            findRandomPositionToWalkTo();
+                        }
+                    }
+                }
+            }
+            else if (raider.blockPosition().distSqr(targetBlock) < 25)
+            {
+                findRandomPositionToWalkTo();
```
Comment (someaddons) [inline src/main/java/com/minecolonies/coremod/entity/mobs/aitasks/RaiderWalkAI.java:127]: should probably set a short walk timeout here aswell,  walkTimer = raider.level.getGameTime() + TICKS_SECOND * 30;
or so

File: src/main/java/com/minecolonies/coremod/entity/mobs/aitasks/RaiderWalkAI.java (hunk 1 above)
Comment (Raycoms) [reply src/main/java/com/minecolonies/coremod/entity/mobs/aitasks/RaiderWalkAI.java:88]: good point

File: src/main/java/com/minecolonies/coremod/entity/mobs/aitasks/RaiderWalkAI.java (hunk 2 above)
Comment (Raycoms) [reply src/main/java/com/minecolonies/coremod/entity/mobs/aitasks/RaiderWalkAI.java:127]: fair, but probably only if walkTimer is > than that

File: src/main/java/com/minecolonies/coremod/entity/pathfinding/pathjobs/PathJobRandomPos.java [hunk 3]
```diff
@@ -163,6 +168,55 @@ protected double getNodeResultScore(@NotNull final MNode n)
...
+      final boolean isSwimming,
+      final boolean onPath,
+      final boolean onRails,
+      final boolean railsExit,
+      final boolean swimStart,
+      final boolean corner,
+      final BlockState state,
+      final BlockPos blockPos)
+    {
+        double cost = Math.sqrt(dPos.getX() * dPos.getX() + dPos.getY() * dPos.getY() + dPos.getZ() * dPos.getZ());
```
Comment (someaddons) [inline src/main/java/com/minecolonies/coremod/entity/pathfinding/pathjobs/PathJobRandomPos.java]: what is this for?

File: src/main/java/com/minecolonies/coremod/entity/mobs/aitasks/RaiderWalkAI.java (hunk 2 above)
Comment (someaddons) [reply src/main/java/com/minecolonies/coremod/entity/mobs/aitasks/RaiderWalkAI.java:127]: think you can use it anyways, walk timer is intended as timeout for the current walking action, so you set the one you want as timeout for in building walk

File: src/main/java/com/minecolonies/coremod/entity/pathfinding/pathjobs/PathJobRandomPos.java (hunk 3 above)
Comment (Raycoms) [reply src/main/java/com/minecolonies/coremod/entity/pathfinding/pathjobs/PathJobRandomPos.java]: Removed door cost from random pathing, this made them avoid doors, soooo much, they would never enter the bedrooms =D

**Review by someaddons — APPROVED** [review]

**Pull request merged**